File: Python/valid_palindrome.py
import re


def is_palindrome(str):
    str.lower()
    clean_str = []
    test = r"[a-zA-Z]"
    for char in str:
        if bool(re.match(test, char)):
            clean_str.append(char)
    print(clean_str)
    reverse_string = []
    for char in clean_str:
        shift = clean_str.pop()
        reverse_string.append(shift)

    print(reverse_string)


is_palindrome("A man, a plan, a canal: Panamaz")

File: JavaScript/validPalindrome.js
// Steps: 
// 1. Remove all non-letters, make everything lowercase
// 2. Compare if the original string minus symbols & lowercase === reverse string minus symbols & lowercase


// First go around (with internet assistance)
var isPalindrome = function(s) {
    let cache = s.split('')
    let output = []
    const check = new RegExp('[a-zA-Z0-9]');
    cache.forEach((letter, index) => {
        console.log(letter)
        if(check.test(letter)){
            output.push(letter)
        }
    });
    let forwardString = output.join('').toLowerCase()
    let reverseString = output.reverse().join('').toLowerCase()
    if(forwardString === reverseString){
        return true
    }
    return false
};

let input = "A man, a plan, a canal: Panama"

console.log(isPalindrome(input))

